app/employee-app/backend/modules/database/db_functions.bal

```diff
@@ -0,0 +1,440 @@
+        return error(customError);
+    }
+    return executionResults.lastInsertId.ensureType();
+}
+
+# Retrieve career function data from HRIS.
+#
+# + filter - Criteria to filter the data  
+# + limit - Number of records to retrieve
+# + offset - Number of records to offset
+# + return - List of career functions
+public isolated function getCareerFunctions(CareerFunctionFilter filter, int 'limit, int offset)
+    returns CareerFunction[]|error {
+
+    CareerFunction[] careerFunctions = [];
+    stream<DBCareerFunction, sql:Error?> resultStream = databaseClient->query(getCareerFunctionsQuery(
+        filter = filter, 'limit = 'limit, offset = offset));
+    error? iterateError = from DBCareerFunction careerFunction in resultStream
+        do {
+            string resDesignations = careerFunction.designations ?: "[]";
+            Designation[]|error designations = resDesignations.fromJsonStringWithType();
+
+            if designations is error {
+                string errorMsg = string `An error occurred when retrieving designations data of ${
+                    careerFunction.careerFunction
+                } !`;
+                log:printError(errorMsg, designations);
+                return error(errorMsg);
+            }
+            careerFunctions.push({
+                id: careerFunction.id,
+                careerFunction: careerFunction.careerFunction,
+                designations: designations
+            });
+        };
+    if iterateError is sql:Error {
+        string errorMsg = string `An error occurred when retrieving career function data!`;
+        log:printError(errorMsg, iterateError);
+        return error(errorMsg);
+    }
+    return careerFunctions;
+}
+
+# Retrieve compensation data of a given filter.
+#
+# + filter - Filter to retrieve the compensation data
+# + return - Compensation data or error
+public isolated function getCompensation(string filter) returns types:CompensationEmail|error {
+
+    CompensationEmail|error compensationRecord = databaseClient->queryRow(getCompensationQuery(filter));
+    if compensationRecord is error {
+        if compensationRecord is sql:NoRowsError {
+            return error(string `No matching compensation data found for ${filter}!`);
+        }
+        string customError = string `Error occurred while retrieving the compensation data for ${filter}!`;
+        log:printError(customError, compensationRecord);
+        return error(customError);
+    }
+
+    types:Compensation[]|error compensation = compensationRecord.keyValuePairs.fromJsonStringWithType();
+    if compensation is error {
+        string customError = string `Error occurred while converting compensation data for ${filter}!`;
+        log:printError(customError, compensation);
+        return error(customError);
+    }
+    if compensation.length() == 0 {
+        error customError = error(string `No matching compensation data found for ${filter}`);
+        return customError;
```

Comment:
Don't need a separate variable here? Can just use the error constructor in the return statement?

Reply:
Check other places too, when you just create the value and return it.